addons/account_facturx/models/account_move.py

```diff
@@ -35,22 +36,46 @@ def format_date(dt):
 
         def format_monetary(number, currency):
             # Format the monetary values to avoid trailing decimals (e.g. 90.85000000000001).
+            if currency.is_zero(number):  # Ensure that we never return -0.0
+                number = 0.0
             return float_repr(number, currency.decimal_places)
 
         # Create file content.
+        seller_siret = hasattr(self.company_id, 'siret') and self.company_id.siret
+        buyer_siret = hasattr(self.commercial_partner_id, 'siret') and self.commercial_partner_id.siret
         template_values = {
             'record': self,
             'format_date': format_date,
             'format_monetary': format_monetary,
+            'balance_multiplicator': -1 if self.is_inbound() else 1,
             'invoice_line_values': [],
+            'SellerSpecifiedLegalOrganization': seller_siret or self.company_id.company_registry,
+            'BuyerSpecifiedLegalOrganization': buyer_siret,
         }
 
         # Tax lines.
-        aggregated_taxes_details = {line.tax_line_id.id: {
-            'line': line,
-            'tax_amount': -line.amount_currency if line.currency_id else -line.balance,
-            'tax_base_amount': 0.0,
-        } for line in self.line_ids.filtered('tax_line_id')}
+        # The old system was making one total "line" per tax in the xml, by using the tax_line_id.
+        # The new one is making one total "line" for each tax category and rate group.
+        aggregated_taxes_details = self._prepare_edi_tax_details(
+            grouping_key_generator=lambda tax_values: {
+                'CategoryCode': tax_values['tax_id']._get_tax_category_code(self.commercial_partner_id, self.company_id),
+                'amount': tax_values['tax_id'].amount
+            }
+        )['tax_details']
+
+        # Map the new keys from the backported _prepare_edi_tax_details into the old ones for compatibility with the old
+        # template.
+        for tax_detail in aggregated_taxes_details.values():
+            tax_detail['tax_base_amount'] = tax_detail['base_amount_currency']
+            tax_detail['tax_amount'] = tax_detail['tax_amount_currency']
+
+            # The old template would get the amount from a tax line given to it, while
+            # the new one would get the amount from the dictionary returned by _prepare_edi_tax_details directly.
+            # As the line was only used to get the tax amount, giving it any line with the same amount will give a correct
+            # result even if the tax line doesn't make much sense as this is a total that is not linked to a specific tax.
+            # I don't have a solution for 0% taxes, we will give an empty line that will allow to render the xml, but it
+            # won't be completely correct. (The RateApplicablePercent will be missing for that line)
+            tax_detail['line'] = self.line_ids.filtered(lambda l: l.tax_line_id and l.tax_line_id.amount == tax_detail['amount'])[:1]
```

Comment:
only `tax_line_id` is used here so `self.line_ids.filtered(lambda x: x.tax_line_id == tax_detail['tax'])` should be enough

Reply:
Because we are using the grouped tax details, we don't have a tax here. We could access the list of taxes from "group_tax_details" inside the dict, but we may as well use my solution and pick the first line that fits.
In the end, the amount is what really matter so to me it makes sense to check it.